Detectors/TPC/qc/src/PID.cxx

```diff
@@ -8,50 +8,99 @@
+
+  mHist2D.emplace_back("hdEdxVsPhi", "dEdx (a.u.) vs #phi (rad); #phi (rad); dEdx (a.u.)", 180, -M_PI, M_PI, 300, 0, 300); //| mHist2D[0]
+  mHist2D.emplace_back("hdEdxVsTgl", "dEdx (a.u.) vs tan#lambda; tan#lambda; dEdx (a.u.)", 60, -2, 2, 300, 0, 300);        //| mHist2D[1]
+  mHist2D.emplace_back("hdEdxVsncls", "dEdx (a.u.) vs ncls; ncls; dEdx (a.u.)", 80, 0, 160, 300, 0, 300);                  //| mHist2D[2]
+
+  const auto logPtBinning = helpers::makeLogBinning(30, 0.1, 10);
+  if (logPtBinning.size() > 0) {
+    mHist2D.emplace_back("hdEdxVsp", "dEdx (a.u.) vs p (G#it{e}V/#it{c}); p (G#it{e}V/#it{c}); dEdx (a.u.)", logPtBinning.size() - 1, logPtBinning.data(), 300, 0, 300); //| mHist2D[3]
+  }
+  //mHist2D.emplace_back("hdedxVsphiMIPA","; #phi (rad); dedx (a.u.)", 180,-M_PI,M_PI,25,35,60);  //| mHist2D[4]
+  //mHist2D.emplace_back("hdedxVsphiMIPC","; #phi (rad); dedx (a.u.)", 180,-M_PI,M_PI,25,35,60);  //| mHist2D[5]
 }
 
+//______________________________________________________________________________
 void PID::resetHistograms()
 {
   for (auto& hist : mHist1D) {
     hist.Reset();
   }
+  for (auto& hist : mHist2D) {
+    hist.Reset();
+  }
 }
 
+//______________________________________________________________________________
 bool PID::processTrack(o2::tpc::TrackTPC const& track)
 {
   // ===| variables required for cutting and filling |===
   const auto p = track.getP();
-  const auto dEdx = track.getdEdx().dEdxTotTPC;
-
-  // ===| cuts |===
-  // hard coded cuts. Should be more configural in future
-  if (std::abs(p - 0.5) > 0.05) {
-    return false;
-  }
-
-  if (dEdx > 70) {
-    return false;
-  }
+  const auto dEdxTot = track.getdEdx().dEdxTotTPC;
+  const auto dEdxMax = track.getdEdx().dEdxMaxTPC;
+  const auto phi = track.getPhi();
+  const auto tgl = track.getTgl();
+  const auto snp = track.getSnp();
+  const auto nclusters = track.getNClusterReferences();
 
   // ===| histogram filling |===
-  mHist1D[0].Fill(dEdx);
+  mHist1D[0].Fill(nclusters);
+  mHist1D[1].Fill(dEdxTot);
+  mHist1D[2].Fill(dEdxMax);
+  mHist1D[3].Fill(phi);
+  mHist1D[4].Fill(tgl);
+  mHist1D[5].Fill(snp);
+
+  mHist2D[0].Fill(phi, dEdxTot);
+  mHist2D[1].Fill(tgl, dEdxTot);
+  mHist2D[2].Fill(nclusters, dEdxTot);
+  mHist2D[3].Fill(p, dEdxTot);
 
   return true;
 }
+
+//______________________________________________________________________________
+void PID::dumpToFile(const std::string filename)
+{
+  auto f = std::unique_ptr<TFile>(TFile::Open(filename.c_str(), "recreate"));
+  f->WriteObject(&mHist1D[0], "nclusters");
+  f->WriteObject(&mHist1D[1], "dEdxTot");
+  f->WriteObject(&mHist1D[2], "dEdxMax");
+  f->WriteObject(&mHist1D[3], "phi");
+  f->WriteObject(&mHist1D[4], "tgl");
+  f->WriteObject(&mHist1D[5], "snp");
+  f->WriteObject(&mHist2D[0], "dEdxVsphi");
+  f->WriteObject(&mHist2D[1], "dEdxVstgl");
+  f->WriteObject(&mHist2D[2], "dEdxVsnclusters");
+  f->WriteObject(&mHist2D[3], "dEdxVsp");
+  f->Close();
```

Comment:
Also here, why not
```c++
for (auto& hist : mHist1D) {
  f->WriteObject(&hist, hist.GetName());
}
```
also for 2D. Otherwise you would need to modify the function whenever you add histograms.

Or even better if this works for vectors of histograms would be to write the whole collection:
```c++
f->Write(&mHist1D, "Hist1D");
f->Write(&mHist2D, "Hist2D");
```
To be tried out.

Reply:
For now I am going with writing the single objects to file in a loop as you suggested. I tried writing the whole vectors and they are written to file but 

1. the root browser can not properly handle the vectors so there is no histogram to look at
2. I did not yet manage to draw the histograms from the root shell but that may be a user problem ;)

I will further investigate that.

Reply:
Ok, then lets keep it with the individual histograms written as it is.